cfme/common/vm.py

```diff
@@ -124,6 +123,10 @@ def factory(cls, vm_name, provider, template_name=None, template=False):
         """
         # Ensure the classes are loaded:
         import cfme.cloud.instance  # NOQA
+        from cfme.cloud.instance.azure import AzureInstance  # NOQA
```

Comment:
Found that vm.factory calls weren't selecting the correct class for cloud instances. Since these weren't imported their classes aren't registered.